__tests__/app.test.js

```diff
@@ -83,6 +83,57 @@ describe("app.js GET requests", () => {
         })
     })
 
+    describe("GET /api/articles/:article_id", () => {
+
+        //Create random number within range of test data values for parametric endpoint testing
+        let article_id = Math.floor(Math.random()*(15-1))+1;
+
+    //Functionality tests
+        test("Status: 200 should return a single object", () => {
+            return supertest(app)
+            .get(`/api/articles/${article_id}`)
+            .then((result) => {
+                expect(200);
+                expect(result.body).toBeInstanceOf(Object);
+                expect(Array.isArray(result.body)).toBe(false);
+            })
+        })
+        test("returned article object has expected properties in expected formats", () => {
+            return supertest(app)
+            .get(`/api/articles/${article_id}`)
+            .then((result) => {
+                console.log(result.body);
+                expect(200);
+                expect(result.body).toHaveProperty("author", expect.any(String));
+                expect(result.body).toHaveProperty("title", expect.any(String));
+                expect(result.body).toHaveProperty("article_id", expect.any(Number));
+                expect(result.body).toHaveProperty("body", expect.any(String));
+                expect(result.body).toHaveProperty("topic", expect.any(String));
+                expect(result.body).toHaveProperty("created_at", expect.any(String));
+                expect(result.body).toHaveProperty("votes", expect.any(Number));
+                expect(result.body).toHaveProperty("article_img_url", expect.any(String));
+            })
+        })
```

Comment:
I can see what you're doing here but it's a little too much, here's some feedback:

- No need to generate a random number
- I recommend you just do a request test to an id number you know exists, i.e. article 3? `/api/articles/3`
- When we assert the response, we need to check that the contents are exactly what we expect them to be, instead of checking the 'shape' of the returned data like you have. So for example, if you send a request to `/api/articles/3` does the response body contain an object which has an `article_id: 3`, is the title the exact title you expect it to be (instead of checking it's a string), and the same for all other keys on the object.

Please refactor this test with the feedback above, removing the random number generation. Your test should be pretty specific of: if you ask for article 3, do you get the exact article 3 contents back in the response? It can just be 1 test, as it's an integration test.